core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -200,7 +200,7 @@ public Plan plan(Analysis analysis)
 
     public Plan plan(Analysis analysis, Stage stage)
     {
-        return plan(analysis, stage, analysis.getStatement() instanceof Explain || isCollectPlanStatisticsForAllQueries(session));
+        return plan(analysis, stage, analysis.getStatement() instanceof ExplainAnalyze || isCollectPlanStatisticsForAllQueries(session));
```

Comment:
please add a test (see `io.trino.tests.TestDistributedEngineOnlyQueries#testExplainAnalyze`)

Reply:
> please add a test (see `io.trino.tests.TestDistributedEngineOnlyQueries#testExplainAnalyze`)

ok，I will do it

Reply:
> please add a test (see `io.trino.tests.TestDistributedEngineOnlyQueries#testExplainAnalyze`)

I had add test case in io.trino.tests.TestDistributedEngineOnlyQueries#testExplainAnalyze,can you review again?